lang/printer/src/types.rs

```diff
@@ -21,34 +26,69 @@ pub type Builder<'a> = pretty::DocBuilder<'a, Alloc<'a>, Anno>;
 /// Operator precedences
 pub type Precedence = u32;
 
-pub trait Print<'a> {
-    fn print(&'a self, cfg: &PrintCfg, alloc: &'a Alloc<'a>) -> Builder<'a> {
+pub trait Print {
```

Comment:
Maybe add some documentation to explain which methods should usually be implemented (`print` or `print_prec`).